testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java

```diff
@@ -485,4 +511,177 @@ private void testSpecialCharacterColumnNamesWithColumnMappingMode(String mode)
+    public void testSupportedNonPartitionedColumnMappingWrites(String columnMappingMode, boolean statsAsJsonEnabled)
+    {
+        String tableName = "test_dl_dml_column_mapping_mode_" + columnMappingMode + randomNameSuffix();
+
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                " (a_number INT, a_string STRING, array_col ARRAY<STRUCT<array_struct_element: STRING>>, nested STRUCT<field1: STRING>)" +
+                " USING delta " +
+                " LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'" +
+                " TBLPROPERTIES (" +
+                " delta.checkpointInterval = 1, " +
+                " delta.checkpoint.writeStatsAsJson = " + statsAsJsonEnabled + ", " +
+                " delta.checkpoint.writeStatsAsStruct = " + !statsAsJsonEnabled + ", " +
+                " delta.columnMapping.mode='" + columnMappingMode + "')");
+
+        try {
+            String queryTrino = "a_number, a_string, array_col[1].array_struct_element, nested.field1";
+            String queryDelta = "a_number, a_string, array_col[0].array_struct_element, nested.field1";
+
+            onTrino().executeQuery("INSERT INTO delta.default." + tableName +
+                    " VALUES (1, 'first value', ARRAY[ROW('nested 1')], ROW('databricks 1'))," +
+                    "        (2, 'two', ARRAY[ROW('nested 2')], ROW('databricks 2'))");
+            onDelta().executeQuery("INSERT INTO default." + tableName +
+                    " VALUES (3, 'third value', array(struct('nested 3')), struct('databricks 3'))," +
+                    "        (4, 'four', array(struct('nested 4')), struct('databricks 4'))");
+            assertDeltaTrinoTableEquals(tableName, queryTrino, queryDelta, ImmutableList.of(
+                    row(1, "first value", "nested 1", "databricks 1"),
+                    row(2, "two", "nested 2", "databricks 2"),
+                    row(3, "third value", "nested 3", "databricks 3"),
+                    row(4, "four", "nested 4", "databricks 4")));
+
+            assertThat(onTrino().executeQuery("SHOW STATS FOR delta.default." + tableName))
+                    .containsOnly(ImmutableList.of(
+                            row("a_number", null, null, 0.0, null, "1", "4"),
+                            row("a_string", null, null, 0.0, null, null, null),
+                            row("array_col", null, null, null, null, null, null),
+                            row("nested", null, null, null, null, null, null),
+                            row(null, null, null, null, 4.0, null, null)));
+
+            onTrino().executeQuery("UPDATE delta.default." + tableName + " SET a_number = a_number + 10 WHERE a_number in (3, 4)");
+            onDelta().executeQuery("UPDATE default." + tableName + " SET a_number = a_number + 20 WHERE a_number in (1, 2)");
+            assertDeltaTrinoTableEquals(tableName, queryTrino, queryDelta, ImmutableList.of(
```

Comment:
Can you pls add a test for `MERGE` as well?

Reply:
added